src/libs/Network/SequentialQueue.ts

```diff
@@ -160,8 +163,35 @@ function isRunning(): boolean {
 NetworkStore.onReconnection(flush);
 
 function push(request: OnyxRequest) {
+    // identify and handle any existing requests that conflict with the new one
+    const requests = PersistedRequests.getAll().filter((persistedRequest) => persistedRequest !== currentRequest);
+    let hasConflict = false;
+    const {getConflictingRequests, handleConflictingRequest, shouldIncludeCurrentRequestOnConflict} = request;
+    if (getConflictingRequests) {
+        // Identify conflicting requests according to logic bound to the new request
+        const conflictingRequests = getConflictingRequests(requests);
+        hasConflict = conflictingRequests.length > 0;
+
+        // Delete the conflicting requests
+        PersistedRequests.bulkRemove(conflictingRequests);
+
+        // Allow the new request to perform any additional cleanup for a cancelled request
+        if (handleConflictingRequest) {
+            for (const conflictingRequest of conflictingRequests) {
+                handleConflictingRequest(conflictingRequest);
+            }
+        }
+    }
+
+    // Don't try to serialize conflict resolution functions
+    delete request.getConflictingRequests;
+    delete request.handleConflictingRequest;
+    delete request.shouldIncludeCurrentRequestOnConflict;
+
     // Add request to Persisted Requests so that it can be retried if it fails
-    PersistedRequests.save(request);
+    if (!hasConflict || !shouldIncludeCurrentRequestOnConflict) {
```

Comment:
Because we flip the condition from `shouldExcludeCurrentRequestOnConflict` to `shouldIncludeCurrentRequestOnConflict`, we should remove the negation too.

```suggestion
    if (!hasConflict || shouldIncludeCurrentRequestOnConflict) {
```

I think we also need to set the default value for `shouldIncludeCurrentRequestOnConflict` to true.

Reply:
@bernhardoj  applying those changes doesn't seems to fix the issue I mentioned above:

https://github.com/Expensify/App/assets/16502320/2e1942d4-7348-4941-b11d-b51b5e1c9342

Reply:
Oh, the suggestion is to have the name match what it does. `shouldIncludeCurrentRequestOnConflict` tells whether we should include the current request even if there is a conflict. If we negate it (!), then it's read as, don't include the current request when there is a conflict, but inside the `if`, we include the request.

Reply:
I see, the issue above also happened on Staging, it doesn't relate to this PR.

Reply:
I think the problem is that the name is confusing. I see why too - right now `shouldIncludeCurrentRequestOnConflict=true` means something like:

"if this request conflicts with other serialized requests, cancel it along with those other requests"

and `shouldIncludeCurrentRequestOnConflict=false` means something like:

"if this request conflicts with other serialized requests, cancel those other requests but still send this one"

Reply:
I went ahead and renamed this param to `shouldSkipThisRequestOnConflict`, which feels more clear to me:

- If `hasConflict && shouldSkipThisRequestOnConflict` then the request is skipped
- if `hasConflict && !shouldSkipThisRequestOnConflict`, then the request is not skipped